### PR TITLE
WebAssembly: Add PNSE for System.Net.Mail

### PR DESCRIPTION
**Pull request by akoeplinger**
It's not supported on WebAssembly so throw PlatformNotSupportedException.

### CODE REVIEW

File: src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs [hunk 1]
```diff
@@ -606,7 +336,7 @@ public void SendAsync(MailMessage message, object? userToken)
                 throw new ObjectDisposedException(GetType().FullName);
             }
 
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, message, userToken, _transport);
```
Comment (akoeplinger) [inline src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs:609]: most of this PR is just moving code but this piece isn't so I want to call it out: I didn't see a great way to split this method up given we only need `_transport` here for the network case.

I don't think we need to pass `_transport` to NetEventSource given it's already captured by the `NetEventSource.Associate(this, _transport);` in the `Initialize()` method so I removed it, however I'd appreciate a second look from someone more familiar with this 😄

**Comment by stephentoub** [comment]
> It allows using the code paths that write a mail message to the file system

Will anyone actually use this in the browser?  I'm wondering if we should instead just add something along the lines of:
```XML
<GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SmtpNetworkDeliveryNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
```
to the project file.

**Comment by akoeplinger** [comment]
@stephentoub I can imagine someone wanting to create mail messages in the browser context and with this change the unit tests that create one in the file system work.

**Comment by stephentoub** [comment]
> I can imagine someone wanting to create mail messages in the browser context and with this change the unit tests that create one in the file system work.

Can you share what such a scenario looks like?  What does someone do with a mail message they create in the browser's file system?

**Comment by akoeplinger** [comment]
> Can you share what such a scenario looks like? What does someone do with a mail message they create in the browser's file system?

Pass it to another library, inspect it locally, send it to an API, .... I don't have a super important use case in mind but it did work in the mono/mono WASM implementation and it was quick to make it work in dotnet/runtime so why not? 😄

**Comment by stephentoub** [comment]
> it was quick to make it work in dotnet/runtime so why not?

Simply because it adds complication/complexity, and it seems a bit far-fetched.  But if you really believe it adds value, ok.

**Comment by akoeplinger** [comment]
@stephentoub I changed the PR to throw PNSE instead.

**Review by marek-safar — APPROVED** [review]

**Review by stephentoub — COMMENTED** [review]
LGTM.  Thanks.

**Pull request merged**